### PR TITLE
helm-chart added

### PR DESCRIPTION
**Pull request by tg44**
Based on the docker-compose file and the [PR to the main repo](https://github.com/helm/charts/pull/5168/files).

What I did;
 - templated out the boilerplate of deployment, service, and pvc configurations
 - removed boilerplatey configs with template hacks (like the command and env params in the `memcached.yaml`)
 - made a somewhat clean but flexible `values.yaml` (resources, affinity, etc can be freely added)

What is missing;
 - we should "host" the chart, for this we should use CI to push artifacts back to a separate branch
   - something like [this method](https://github.com/kmzfs/helm-repo-in-github) could work with CircleCI, and GH-actions too
   - or [this method](https://dev.to/jamiemagee/how-to-host-your-helm-chart-repository-on-github-3kd) seems more professional, but also it is harder to keep up to date later, and probably hard to port to CircleCI
 - installations to check if it actually works for others too :)
 - it needs some ingress tweaking, now it throws warrnings bcs of the deprecated apiVersion (not a big deal right now)
 - rewrite the README.md

### CODE REVIEW

**Comment by flavio-prado** [comment]
Hello,
How should I configure if I don't want extra memcached, postgres, rabbit and redis in my cluster and instead want to use the ones that I already have? Is there an option to disable it?

**Comment by flavio-prado** [comment]
What do you think about creating a service account or using an existing one?
In my case, for example, I use external redis on aws and my service account to give access to my aws redis service.

**Comment by tg44** [comment]
It was an all-in-one chart (when I wrote it month ago we needed a full setup). We could "if" the dependencies one-by-one if this is a feature request, but it can't disable dependent services right now.

About service accounts; I have no idea. If we add the option to disable dependent services, we should add another option to connect to them. I think the easiest way to provide a connection string/url and/or user+pass. BTW I think if you have half of the needed infra, you should cook up a custom yaml for the remaining. In my understanding helm charts mostly useful if you want to pack a multiservice application into a single install.

File: kubernetes/chart/zulip/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+{{- end }}
+{{- else }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "zulip.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
+  kubectl port-forward $POD_NAME 8080:80
+{{- end }}
+
+2. To create a realm so you can sign in:
+
+  kubectl -n {{ .Release.Namespace }} exec -it $(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "zulip.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}") -c zulip /home/zulip/deployments/current/manage.py generate_realm_creation_link
```
[the remark this answers is not in the document]
Comment (mrtndwrd) [reply kubernetes/chart/zulip/templates/NOTES.txt:14]: I'm not part of the Zulip team, so I don't think I have the rights to amend the commit. This also means I wouldn't have the rights to merge this if you would amend the commit.

That said, I think I'll have to make some more changes before the helm chart would be usable for my purpose. I started a conversation in [#provision-help/K8 and Helm](https://chat.zulip.org/#narrow/stream/21-provision-help/topic/K8.20and.20Helm) about this too.

I'm not sure yet if my organization is continuing with Zulip, but if we do, would it be OK with you if I would make a new PR based on yours? Or would you like to be more involved than that?

[the remark this answers is not in the document]
Comment (tg44) [reply kubernetes/chart/zulip/templates/NOTES.txt:14]: You can start a new PR on top of my commit, and I will close this in favor of that.

[the remark this answers is not in the document]
Comment (varac) [reply kubernetes/chart/zulip/templates/NOTES.txt:14]: Done and merged (#325): https://github.com/zulip/docker-zulip/tree/main/kubernetes/chart/zulip.
@tg44 Please close this PR

**Pull request closed without merging**